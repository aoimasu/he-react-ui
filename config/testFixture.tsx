import * as React from 'react';
import createTestContext from 'react-cosmos-test/enzyme';

type Fixture = {
  component: React.ComponentType<any>;
  name?: string;
};

function testOneFixture(fixture: Fixture, variant: string = 'a') {
  const { mount, getWrapper } = createTestContext({ fixture });
  const { component } = fixture;
  const { displayName, name } = component as any;
  mount();
  test(`<${displayName || name} /> rendered correctly with ${fixture.name ||
    variant} fixture`, () => expect(getWrapper()).toMatchSnapshot());
}

// TODO: use real type guard functions to discriminate one fixture vs an array of them

export default function testAllFixtures(fixtures: any, variant?: string) {
  if (!('forEach' in fixtures)) {
    testOneFixture(fixtures as Fixture, variant);
  } else if (typeof fixtures.forEach === 'function') {
    (fixtures as Fixture[]).forEach(it => testOneFixture(it));
  }
}
