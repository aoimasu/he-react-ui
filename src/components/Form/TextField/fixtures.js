// @flow
import React from 'react';
import Input from '.';

export default [
  {
    name: 'Default',
    formWrapper: { value: '' },
    component: Input,
    props: {
      id: 'demo',
      name: 'demo',
      label: 'First Name',
    },
  },

  {
    name: 'small',
    formWrapper: { value: '' },
    component: Input,
    props: {
      id: 'demo',
      name: 'demo',
      label: 'First Name',
      small: true,
    },
  },

  {
    name: 'smallNoLabel',
    formWrapper: { value: '' },
    component: Input,
    props: {
      id: 'demo',
      name: 'demo',
      small: true,
    },
  },

  {
    name: 'password',
    formWrapper: { value: '' },
    component: Input,
    props: {
      id: 'demo',
      name: 'demo',
      label: 'Password',
      password: true,
    },
  },

  {
    name: 'noLabel',
    formWrapper: { value: '' },
    component: Input,
    props: {
      id: 'demo',
      name: 'demo',
    },
  },

  {
    name: 'Disabled',
    component: Input,
    props: {
      id: 'demo',
      name: 'demo',
      label: 'First Name',
      disabled: true,
    },
  },

  {
    name: 'Error',
    component: Input,
    props: {
      id: 'demo',
      name: 'demo',
      label: 'First Name',
      error: "Something isn't right",
    },
  },

  {
    name: 'InvalidMarker',
    component: Input,
    props: {
      id: 'demo',
      name: 'demo',
      label: 'First Name',
      error: "Something isn't right",
      marker: true,
      value: 'Bruce',
    },
  },

  {
    name: 'ValidMarker',
    component: Input,
    props: {
      id: 'demo',
      name: 'demo',
      label: 'First Name',
      marker: true,
      value: 'Bruce',
    },
  },

  {
    name: 'Value',
    component: Input,
    props: {
      id: 'demo',
      name: 'demo',
      label: 'First Name',
      value: 'Oscar',
    },
  },

  {
    name: 'With helper',
    component: Input,
    props: {
      id: 'demo',
      name: 'demo',
      label: 'First Name',
      helper: <div>Test HTML</div>,
    },
  },

  {
    name: 'With verified',
    component: Input,
    props: {
      id: 'demo',
      name: 'demo',
      label: 'First Name',
      verified: <div>Verify</div>,
    },
  },
];
