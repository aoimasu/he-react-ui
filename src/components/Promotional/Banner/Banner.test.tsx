import fixtures from './Banner.fixture';
import testFixture from '../../../../config/testFixture';

testFixture(fixtures);
