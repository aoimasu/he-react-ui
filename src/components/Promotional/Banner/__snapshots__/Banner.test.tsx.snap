// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner /> rendered correctly with a fixture 1`] = `
<Banner
  title="The Heading"
>
  <div
    className="outer"
  >
    <h1
      className="title"
    >
      The Heading
    </h1>
    <div
      className="content"
    >
      <p>
        This is a paragraph of text made to emulate a quick description of the item being promoted.
      </p>
      <Button
        buttonProps={Object {}}
        color="teal"
        disabled={false}
        icon=""
        keyline={false}
        large={true}
        link={false}
        newWindow={false}
        onClick={[Function]}
        squared={false}
        submit={false}
        withRouter={false}
      >
        <div
          className="buttonContainer large"
        >
          <InnerButton
            buttonProps={Object {}}
            className="button teal"
            newWindow={false}
            onClick={[Function]}
            submit={false}
            withRouter={false}
          >
            <button
              className="button teal"
              onClick={[Function]}
              type="button"
            >
              <div
                className="content"
              >
                Activate now
              </div>
            </button>
          </InnerButton>
        </div>
      </Button>
      <Button
        buttonProps={Object {}}
        color="white"
        disabled={false}
        icon=""
        keyline={true}
        large={true}
        link={false}
        newWindow={false}
        onClick={[Function]}
        squared={false}
        submit={false}
        withRouter={false}
      >
        <div
          className="buttonContainer large"
        >
          <InnerButton
            buttonProps={Object {}}
            className="button white keyline"
            newWindow={false}
            onClick={[Function]}
            submit={false}
            withRouter={false}
          >
            <button
              className="button white keyline"
              onClick={[Function]}
              type="button"
            >
              <div
                className="content"
              >
                Find out more
              </div>
            </button>
          </InnerButton>
        </div>
      </Button>
    </div>
  </div>
</Banner>
`;
